botocore/session.py

```diff
@@ -516,6 +516,10 @@ def get_credentials(self):
         credentials.
 
         """
+        # This enables feature id registration when credentials are set in a session
+        # and that session is used for client creation.
+        if getattr(self._credentials, 'method', None) == 'explicit':
```

Comment:
I'm assuming the getattr is to support people using alternative credentials that don't have a `method` property - do we have a clear use case for this, or is this just out of caution?